sklearn/tree/tests/test_tree.py

```diff
@@ -289,9 +303,13 @@ def test_diabetes_underfit(name, Tree, criterion, max_depth):
         max_features=6, random_state=0
     )
     reg.fit(diabetes.data, diabetes.target)
-    score = mean_squared_error(diabetes.target, reg.predict(diabetes.data))
-    assert score < 60 and score > 0, (
-        f"Failed with {name}, criterion = {criterion} and score = {score}"
+    if criterion == "poisson":
+        loss = mean_poisson_deviance(diabetes.target,
+                                     reg.predict(diabetes.data))
+    else:
+        loss = mean_squared_error(diabetes.target, reg.predict(diabetes.data))
+    assert 0 < loss < max_loss, (
+        f"Failed with {name}, criterion = {criterion} and loss = {loss}"
```

Comment:
IMHO the error message doesn't help much more than what `pytest` would already tell us. I understand it was there before, but we can remove it